src/BIAnalyticsService/RetrieveData/RetrieveData.cs

```diff
@@ -0,0 +1,122 @@
+using System.Net;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.Extensions.Logging;
+using System.Text.Json;
+using NHS.ServiceInsights.Common;
+using NHS.ServiceInsights.Model;
+
+namespace NHS.ServiceInsights.BIAnalyticsService;
+
+public class RetrieveData
+{
+    private readonly ILogger<RetrieveData> _logger;
+    private readonly IHttpRequestService _httpRequestService;
+    public RetrieveData(ILogger<RetrieveData> logger, IHttpRequestService httpRequestService)
+    {
+        _logger = logger;
+        _httpRequestService = httpRequestService;
+    }
+
+    [Function("RetrieveData")]
+    public async Task<HttpResponseData> Run([HttpTrigger(AuthorizationLevel.Anonymous, "get")] HttpRequestData req)
+    {
+        _logger.LogInformation("Request to retrieve episode information has been processed.");
+
+        string episodeId = req.Query["EpisodeId"];
+
+        if (string.IsNullOrEmpty(episodeId))
+        {
+            _logger.LogError("Please enter a valid Episode ID.");
+            var badRequestResponse = req.CreateResponse(HttpStatusCode.BadRequest);
+            return badRequestResponse;
+        }
+
+        var baseUrl = Environment.GetEnvironmentVariable("GetEpisodeUrl");
+        var url = $"{baseUrl}?EpisodeId={episodeId}";
+        _logger.LogInformation("Requesting episode URL: {Url}", url);
+
+        string episodeJson = string.Empty;
+
+        try
+        {
+            var response = await _httpRequestService.SendGet(url);
+
+            if (!response.IsSuccessStatusCode)
+            {
+                _logger.LogError($"Failed to retrieve episode with Episode ID {episodeId}. Status Code: {response.StatusCode}");
+                var errorResponse = req.CreateResponse(response.StatusCode);
+                return errorResponse;
+            }
+
+            episodeJson = await response.Content.ReadAsStringAsync();
+            _logger.LogInformation("Episode data retrieved");
+        }
+
+        catch (Exception ex)
+        {
+            _logger.LogError("Failed to call the GetEpisode Data Service. \nException: {ex}", ex);
+            return req.CreateResponse(HttpStatusCode.InternalServerError);
+        }
+
+        string nhsNumber = "1111111112";
+
+        if (string.IsNullOrEmpty(nhsNumber))
```

Comment:
Given that we've hardcoded the NHS number for now, let's remove this if statement